chain/vm/execution.go

```diff
@@ -0,0 +1,221 @@
+package vm
+
+import (
+	"context"
+	"errors"
+	"os"
+	"strconv"
+	"sync"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/tag"
+
+	"github.com/ipfs/go-cid"
+
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/metrics"
+)
+
+const (
+	DefaultAvailableExecutionLanes = 4
+	DefaultPriorityExecutionLanes  = 2
```

Comment:
As we discussed synchronously, we need to justify this number by looking at the critical path usages in the chain sync subsystem. From my understanding, those are the following, but @magik6k will probably have better ideas.

- Incoming block validation via gossip
- Head coalescer executions.
- Actual deferred execution.

Also note that time-sensitive miner gas estimations and mpool pushes will fall under the "user" section. I'm not sure how much of a problem that could be, probably not much since those daemons won't be facing arbitrary user workloads. But worth confirming.